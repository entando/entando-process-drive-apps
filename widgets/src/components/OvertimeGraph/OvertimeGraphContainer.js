import React, { Component } from 'react';
import { MuiThemeProvider as ThemeProvider } from '@material-ui/core/styles';
import { Paper, Typography, Grid, Box } from '@material-ui/core';

import theme from 'theme';
import BarAreaChart from 'components/OvertimeGraph/BarAreaChart';

class OvertimeGraph extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    return (
      <ThemeProvider theme={theme}>
        <Paper>
          <Box p={2}>
            <Typography variant="h5" gutterBottom>
              Requests Volume
            </Typography>
            <Grid container>
              <Grid item xs={8}>
                <Typography variant="subtitle1">Requests</Typography>
              </Grid>
              <Grid item xs={4}>
                tabs
              </Grid>
            </Grid>
            <Grid container>
              <Grid item xs={8} style={{ height: '250px' }}>
                <BarAreaChart
                  data={[
                    { x: 'Jan 03', bar: 60, area: 8 },
                    { x: 'Jan 06', bar: 45, area: 5 },
                    { x: 'Jan 09', bar: 20, area: 10 },
                    { x: 'Jan 12', bar: 75, area: 15 },
                  ]}
                  legends={{
                    bar: 'Number of Requests',
                    area: 'Bookings Made',
                  }}
                />
              </Grid>
              <Grid item xs={4}>
                details here
              </Grid>
            </Grid>
          </Box>
        </Paper>
      </ThemeProvider>
    );
  }
}

export default OvertimeGraph;
