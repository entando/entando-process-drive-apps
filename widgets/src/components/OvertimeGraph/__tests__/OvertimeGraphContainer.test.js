import React from 'react';
import { render } from '@testing-library/react';

import OvertimeGraphContainer from 'components/OvertimeGraph/OvertimeGraphContainer';

describe('<OvertimeGraphContainer />', () => {
  it('should match snapshot', () => {
    const { container } = render(<OvertimeGraphContainer />);
    expect(container).toMatchSnapshot();
  });
});
