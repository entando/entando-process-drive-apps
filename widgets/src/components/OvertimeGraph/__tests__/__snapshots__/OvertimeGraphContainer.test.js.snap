// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OvertimeGraphContainer /> should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-29"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Requests Volume
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            Requests
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          tabs
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
          style="height: 250px;"
        >
          <div
            class="recharts-responsive-container"
            id="undefined"
            style="width: 100%; height: 100%;"
          >
            <div />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          details here
        </div>
      </div>
    </div>
  </div>
</div>
`;
